Implement configuration file handling

MytsV opened the pull request:
# Purpose

To prepare the app for database implementation. The configuration will make it easy to change the storage path and create shared to-do lists between users. In the future, more parameters may be configured dynamically.

# Summary

Initiated the possibility of reading a user-specific configuration file. Currently, the configuration handler supports only PATH entry, which points to the database directory, and DEBUG. If no config is found in the user's home directory, it is automatically created with a default value.

# Other information

By default, the app's config is stored in the user's home directory under ```.todolor.conf``` name. The default value for the database directory is ```$HOME/.todolor```.

The format of configuration file is as follows:
- content can't be empty
- every non-empty line is a valid key separated by '=' from corresponding value

MytsV committed:
Implement configuration handling

Initiate the possibility to dynamically get app's configuration.
Currently, configuration handler supports only PATH entry, which points to database directory.
The config is user specific, hence os.homedir() calls.
If no config is found in the user's home directory, it is automatically created with default value.

svitbar commented on src/config.js:
```diff
@@ -0,0 +1,44 @@
+const yellowAnsi = '\x1b[33m';
+const defaultAnsi = '\x1b[39m';
+
+const parse = (contents) => {
+  // TODO: Rewrite without regexp
+  const regExp = /PATH=(.*)/;
+  const result = regExp.exec(contents);
+  if (!result || result.length === 0) {
+    throw Error('Corrupted configuration file');
+  } else {
+    return result[0];
+  }
+};
```
Try to rewrite this function cause you can have some issues with newlines.

svitbar commented on src/config.js:
```diff
@@ -0,0 +1,44 @@
+  }
+};
+
+const createConfig = (configPath) => {
+  const databasePath = path.join(os.homedir(), defaultPath);
+  const entry = `PATH=${databasePath}`;
+  fs.writeFileSync(configPath, entry);
+  return databasePath;
+};
```
Decompose this function to make it more readable.

MytsV replied:
I thought about it, the problem is not with readability as such, but with the strange return, which works poorly with the function's name. Will definitely improve it after unit tests are written.

svitbar reviewed (changes requested):
- provide your code with unit tests;
- replace reading the config;
- return some kind of structure Configuration, now you're limited only with one key.

MytsV committed:
Rename getDatabasePath and change its return type

Basically, we changed the purpose of this function. This step
is essential to start developing unit tests.

MytsV committed:
Fix parse function to return correct value

Now it returns only the first capture, not the whole regex match

MytsV committed:
Implement unit tests for configuration handler

Use two new dependencies: sinon for third-party libraries stubs, mock-fs
for imitating file system. Parser test will fail, as handler doesn't read all
configuration fields yet.

MytsV committed:
Rewrite parse function of config handler without RegExp

Additionally, document the function

MytsV committed:
Refactor config.js

Change readConfig's name to read, so that we can call it like config.read().

MytsV committed:
Fix tests to pass in every environment

Remove directory permission test, as it turned out to be platform dependent.
Rename mock home directory, as '/home' name caused unknown collision.

MytsV commented:
Regarding requested changes: everything is done except for "replace reading the config". I think you meant that I shouldn't mix config reading with path extraction. However, this whole function was rewritten, and I believe it looks fine now.

MytsV commented:
**Regarding these shameful force pushes**
The thing is, for some reason mock-fs and GitHub Actions decided not to work well together. Therefore, tests that passed locally and with ```act``` crashed on CI server. There was no way for me to test them (at least, to my knowledge) without sneakily creating a new GitHub repository OR debugging here. I chose the latter, because I value honesty, and I'm not afraid to show the world my embarrassing failures!
What was the deal? Apparently, something collided during the mocking of a file system. It was '/home' directory that was overwritten despite restoring the state. As soon as I changed the name of the default home directory in tests, everything fell into place.
I still don't know the reason for such bug, and, repeatedly, everything works fine for me with ```act```. So, be careful too?

MytsV commented:
Why force? Because I didn't want all the "DEBUG" commits to end up in history.

svitbar reviewed (approved):
Everything looks fine! Thank you for your improvements.
